lib/Sema/SemaBounds.cpp

```diff
@@ -711,14 +764,17 @@ namespace {
       // Bounds of the right-hand side of the assignment
       BoundsExpr *RHSBounds = nullptr;
 
-      // Check that the RHS of the assignment has bounds if the
+      // Check that the value being assigned has bounds if the
       // target of the LHS lvalue has bounds.
       if (LHSType->isCheckedPointerType() ||
-          LHSType->isIntegralOrEnumerationType()) {
+          LHSType->isIntegerType()) {
```

Comment:
What are we doing about EnumerationTypes now? Do we mind that we don't cover them any more?